configure

```diff
@@ -584,7 +584,7 @@ fi
 have_natdynlink=0
 natdynlink=""
 camlp4_dynlink=""
-if [ -f "${ocaml_core_stdlib}/${dynlink_subdir}/dynlink.cmxa" ]; then
+if [ -f "${ocaml_core_stdlib}/dynlink.cmxa" ]; then
```

Comment:
While it has some obvious benefits, this is the problem I have with shellcheck - it becomes too easy to believe.

This regresses #49 - the variable is set here:

https://github.com/ocaml/ocamlfind/blob/c14912adb4c32d1179806b649215ffe49234c37b/configure#L528